src/sif/autocost.cc

```diff
@@ -517,7 +517,15 @@ Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
   }
 
   // TODO: speed_penality hasn't been extensively tested, might alter this in future
-  float speed_penalty = (edge_speed > top_speed_) ? (edge_speed - top_speed_) * 0.05f : 0.0f;
+  float average_historic_speed = edge_speed;
+  // disable dynamic(current/predicted) speed layers and use static
+  // ones(constrained_flow/free_flow/base)
+  if (top_speed_ != kMaxAssumedSpeed && (flow_sources & kDynamicFlowMask)) {
+    // do not pass seconds_of_week not to make speed penalties independent of datetime
+    average_historic_speed = tile->GetSpeed(edge, flow_mask_ & kStaticFlowMask);
+  }
+  float speed_penalty =
+      (average_historic_speed > top_speed_) ? (edge_speed - top_speed_) * 0.05f : 0.0f;
```

Comment:
```suggestion
      (average_historic_speed > top_speed_) ? (average_historic_speed - top_speed_) * 0.05f : 0.0f;
```
?
otherwise, if `edge_speed < top_speed_` we can get negative speed penalty

Reply:
ah, right. Thanks @genadz